doc/dualtor/active_active_hld.md

```diff
@@ -0,0 +1,176 @@
+
+  Link Prober will report 4 possible states:  
+  * LinkProberUnknown: Serves as initial states. This state is also reachable in the case of no ICMP reply is received. 
+  * LinkProberActive: It indicates that LinkMgr receives ICMP replies containing ID of the current ToR.
+  * LinkProberPeerUnknown: It indicates that LinkMgr did not receive ICMP replies containing ID of the peer ToR. Hence, there is a chance that peer ToR’s link is currently down. 
+  * LinkProberPeerAcitve: It indicates that LinkMgr receives ICMP replies containing ID of the peer ToR, or in other words, peer ToR’s links appear to be active.  
+
+  __ICMP Probing Format__  
+  The source MAC will be ToR's SVI mac address. Ethernet destination will be the well-known MAC address. Source IP will be ToR's Loopback IP, destination IP will be SoC's IP address, which will be introduced as a field in minigraph.   
+  ![icmp_format](./image/icmp_format.png)  
+
+  Linkmgrd also adapt TLV (Type-Length-Value) as the encoding schema in payload for additional information elements, including cookie, version, ToR GUID etc.  
+   ![icmp_payload](./image/icmp_payload.png)  
+
+* Link State  
+  When link is down, linkmgrd will receive notification from SWSS based on kernel message from netlink. This notification will be used to determine if ToR is healthy. 
+
+* Admin Forwarding State   
+  ToRs will signal NIC if the link is active / standby, we will call this active / standby state as admin forwarding state. It's up to NIC to determine which link to use if both are active, but it should never choose to use a standby link. This logic provides ToR more control over traffic forwarding.  
+  
+* Acitve-Active State Machine  
+  Active-acitve state transition logics are simplified compared to active-standby. In active-standby, linkmgrd makes mux toggle decisions based on y-cable direction, while for active-active, two links are more independent. Linkmgrd will only make state transition decisions based on healthy indicators. 
+
+  To be more specific, if link prober indicates active AND link state appears to be up, linkmgrd should determine link's forwarding state as active, otherwise, it should be standby.
+
+  ![active_active_self](./image/active_active_self.png) 
+
+  Linkmgrd also provides rescue mechanism when peer can't switch to standby for some reason, i.e. link failures. If link prober doesn't receive peer's heartbeat response AND self ToR is in healthy active state, linkmgrd should determine peer link to be standby. 
+  ![active_active_peer](./image/active_active_peer.png)  
+
+  When control channel is unreachable, ToR won't block the traffic forwarding, but it will periodically check gRPC server's healthiness. It will make sure server side's admin forwarding state aligns with linkmgrd's decision.
+  ![grpc_failure](./image/gRPC_failure.png) 
```

Comment:
Could you update the HLD with the flow from linkmgrd to orchagent to update the nexthops to IPinIP tunnel.

Reply:
Will do!

Reply:
Added.